File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <bits/stdc++.h>
using namespace std;
// DFS로 구현
bool c[201];

void dfs(int current_computer , int n , vector<vector<int>>&computers){
    c[current_computer] = 1;
    
    for(int i = 0; i < n; i ++){
        if(c[i] == 0 && computers[current_computer][i] == 1) dfs(i,n,computers);
    }
}
int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i = 0; i < n; i ++)
    {   
        if(c[i] == 0)
        {dfs(i, n, computers);
        answer ++;}
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(computers, visited, node):
    visited[node] = True  
    for i in range(len(computers)): 
        if computers[node][i] == 1 and not visited[i]: 
            dfs(computers, visited, i)  

def solution(n, computers):
    visited = [False] * n  
    network_count = 0  

    for i in range(n):  
        if not visited[i]:  
            dfs(computers, visited, i)  
            network_count += 1  

    return network_count 
